components/session-based-conditional-auth-functions/src/main/java/org/wso2/carbon/identity/extensions/authenticator/conditional/auth/functions/function/IsWithinSessionLimitFunction.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.util.SessionValidationConstants;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.util.SessionValidationUtil;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Represents javascript function provided in conditional authentication to check if the given user has valid number of
+ * sessions.
+ * The purpose is to perform dynamic authentication selection based on the active session count.
+ */
+public class IsWithinSessionLimitFunction implements IsValidFunction {
+
+    private static final Log log = LogFactory.getLog(IsWithinSessionLimitFunction.class);
+
+    /**
+     * Method to validate user session a given the authentication context and set of required attributes
+     *
+     * @param context Authentication context
+     * @param map     Hash map of attributes required for validation
+     * @return boolean value indicating the validation success/failure
+     * @throws AuthenticationFailedException when exception occurred in session retrieving method
+     */
+    @Override
+    public Boolean validate(JsAuthenticationContext context, Map<String, String> map)
+            throws AuthenticationFailedException {
+
+        boolean state = false;
+        int sessionLimit = getSessionLimitFromMap(map);
+        AuthenticatedUser authenticatedUser = context.getWrapped().getLastAuthenticatedUser();
```

Comment:
Is there any chance context.getWrapped() becoming null if so possible NPE

Reply:
added a try catch block to catch NPE and AuthenticationFailedException thrown